sdk/eventhub/event-hubs/test/eventProcessor.spec.ts

```diff
@@ -484,5 +484,114 @@ describe("Event Processor", function(): void {
       partitionOwnershipList[0].sequenceNumber!.should.equals(checkpoint.sequenceNumber);
       partitionOwnershipList[0].offset!.should.equals(checkpoint.offset);
     });
+
+    it("should receive events from the checkpoint", async function(): Promise<void> {
+      const partitionIds = await client.getPartitionIds();
+
+      // ensure we have at least 2 partitions
+      partitionIds.length.should.gte(2);
+
+      let partitionResultsMap = new Map<string, ReceivedEventData[]>();
+      partitionIds.forEach((id) => partitionResultsMap.set(id, []));
+      let didError = false;
+
+      // The partitionProcess will need to add events to the partitionResultsMap as they are received
+      const factory: PartitionProcessorFactory = (context, checkpointManager) => {
+        return {
+          async processEvents(events: ReceivedEventData[]) {
+            const existingEvents = partitionResultsMap.get(context.partitionId)!;
+            events.forEach((event: ReceivedEventData) => {
```

Comment:
This might be ok since the `InMemoryPartitionManager` is not making any non-blocking calls, but since `updateCheckpoint` returns a promise we should be awaiting it. It might be simpler then to use a for loop instead of `forEach` so you can `await` udpateCheckpoint.

Otherwise if we don't make this change and we updated InMemoryPartitionManager to be non-blocking, we could end up checkpointing events out of order.